src/sonic-config-engine/minigraph.py

```diff
@@ -157,11 +161,94 @@ def parse_png(png, hname):
                             elif node.tag == str(QName(ns, "EndDevice")):
                                 mgmt_dev = node.text
 
-    return (neighbors, devices, console_dev, console_port, mgmt_dev, mgmt_port, port_speeds, console_ports)
+    return (neighbors, devices, console_dev, console_port, mgmt_dev, mgmt_port, port_speeds, console_ports, device_neighbors)
 
+def parse_asic_png(png, asic_name, hostname):
+    neighbors = {}
+    devices = {}
+    port_speeds = {}
+    device_neighbors = []
+    for child in png:
+        if child.tag == str(QName(ns, "DeviceInterfaceLinks")):
+            for link in child.findall(str(QName(ns, "DeviceLinkBase"))):
+                linktype = link.find(str(QName(ns, "ElementType"))).text
+                if linktype != "DeviceInterfaceLink" and linktype != "UnderlayInterfaceLink" and linktype != "DeviceSerialLink":
+                    continue
+                enddevice = link.find(str(QName(ns, "EndDevice"))).text
+                endport = link.find(str(QName(ns, "EndPort"))).text
+                startdevice = link.find(str(QName(ns, "StartDevice"))).text
+                startport = link.find(str(QName(ns, "StartPort"))).text
+                bandwidth_node = link.find(str(QName(ns, "Bandwidth")))
+                bandwidth = bandwidth_node.text if bandwidth_node is not None else None
+                # Chassis internal node is used in multi-asic device or chassis minigraph
+                # where the minigraph will contain the internal asic connectivity and
+                # external neighbor information. The ChassisInternal node will be used to
+                # determine if the link is internal to the device or chassis.
+                chassis_internal_node = link.find(str(QName(ns, "ChassisInternal")))
+                chassis_internal = chassis_internal_node.text if chassis_internal_node is not None else "false"
+
+                # If the link is an external link include the external neighbor
```

Comment:
i suggest to simplify the code here, only parsing the internal links here. if you want to add device neighbor information, you can add it later in the main program.

Reply:
created separate function to parse internal and external links of asic